src/conn/conn_capacity.c

```diff
@@ -0,0 +1,464 @@
+/*
+ * Copyright (c) 2014-2019 MongoDB, Inc.
+ * Copyright (c) 2008-2014 WiredTiger, Inc.
+ *	All rights reserved.
+ *
+ * See the file LICENSE for redistribution information.
+ */
+
+#include "wt_internal.h"
+
+/*
+ * Compute the time in nanoseconds that must be reserved to represent
+ * a number of bytes in a subsystem with a particular capacity per second.
+ */
+#define	WT_RESERVATION_NS(bytes, capacity)			\
+	(((bytes) * WT_BILLION) / (capacity))
+
+/*
+ * The fraction of a second's worth of capacity that will be stolen at a
+ * time. The number of bytes this represents may be different for different
+ * subsystems, since each subsystem has its own capacity per second.
+ */
+#define	WT_STEAL_FRACTION(x)		((x) / 16)
+
+/*
+ * __capacity_config --
+ *	Set I/O capacity configuration.
+ */
+static int
+__capacity_config(WT_SESSION_IMPL *session, const char *cfg[])
+{
+	WT_CAPACITY *cap;
+	WT_CONFIG_ITEM cval;
+	WT_CONNECTION_IMPL *conn;
+	uint64_t total;
+
+	conn = S2C(session);
+
+	WT_RET(__wt_config_gets(session, cfg, "io_capacity.total", &cval));
+	if (cval.val != 0 && cval.val < WT_THROTTLE_MIN)
+		WT_RET_MSG(session, EINVAL,
+		    "total I/O capacity value %" PRId64 " below minimum %d",
+		    cval.val, WT_THROTTLE_MIN);
+
+	if (cval.val != 0) {
+		WT_RET(__wt_calloc_one(session, &conn->capacity));
```

Comment:
It occurs to me, this allocation is 104 bytes, it's not worth the error handling to deal with the possible allocation failure, just add it to the `WT_CONNECTION` structure.

Reply:
Oh, I think `WT_CAPACITY` is a good idea, I'm just saying don't bother allocating the structure, just declare it.

(On second thought, that means all of your `p->` code would become `p.`, which is maybe just annoying pain at this point. Oh, forget it. :)

Reply:
I committed a change for this. It was actually very small.